parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestInputOutputFormatWithBloomFilter.java

```diff
@@ -0,0 +1,249 @@
+      HashMap expectedParameter = new HashMap<String,Long>(){{
+        put("uuid",200L);
+        put("value",50L);
+      }};
+      Assert.assertTrue("Bloom filter parameter should be correct inside Hadoop conf.", expectedParameter.equals(ParquetOutputFormat.getBloomFilterColumnExpectedNDVs(conf)));
+
+      waitForJob(writeJob);
+
+    }
+
+    // Check bloom filter for each block and column
+    Path path = getDataPath(tempFolder);
+
+    ParquetMetadata readFooter = ParquetFileReader.readFooter(conf, path);
+    ParquetFileReader r = new ParquetFileReader(conf,path,readFooter);
+
+    int startBlock = 0;
+    for(BlockMetaData block : readFooter.getBlocks()){
+      int rowCount = (int) block.getRowCount();
+      int endBlock = rowCount + startBlock;
+      int columnIndex = 0;
+      for(ColumnChunkMetaData columnMetaData : block.getColumns()) {
+
+        Long expectedDistinctNumbers = Long.parseLong(EXPECTED_DISTINCT_NUMBERS[columnIndex]);
+
+        // Build bloom filter from the generated data
+        BloomFilter expectedBloomFilter = buildBloomFilter(columnMetaData, startBlock, endBlock,columnIndex, expectedDistinctNumbers);
+
+        // Read bloom filter from the parquet file
+        BloomFilterReader bloomFilterReader = r.getBloomFilterDataReader(block);
+        BloomFilter bloomFilter = bloomFilterReader.readBloomFilter(columnMetaData);
+
+        Assert.assertTrue("BloomFilter bitset should be the same", bloomFilter.equals(expectedBloomFilter));
+
+        columnIndex++;
+      }
+      startBlock = endBlock;
+    }
+
+  }
+
+  private String generateData(){
+    StringBuilder data = new StringBuilder();
+    for(int i =0; i< 200; i++) {
+      String uuid = UUID.randomUUID().toString();
+      Random random = new Random();
+      int value = random.nextInt(50);
+      data.append(uuid).append(":").append(Integer.toString(value)).append("_");
+    }
+    return StringUtils.chop(data.toString());
+
+
+  }
+
+  private BloomFilter buildBloomFilter(ColumnChunkMetaData columnMetaData, int startBlock,int endBlock, int columnIndex, long expectedDistinctNumbers) {
+
+    String[] tuples = this.FILE_CONTENT.split("_");
+
+    int optimalNumOfBits = BlockSplitBloomFilter.optimalNumOfBits(expectedDistinctNumbers, FPP);
+    BloomFilter expectedBloomFilter = new BlockSplitBloomFilter(optimalNumOfBits / 8);
+
+    for (int i = startBlock; i < endBlock; i += 1) {
+      String[] elements = tuples[i].split(":");
+      if (columnMetaData.getPrimitiveType().getPrimitiveTypeName().equals(BINARY)) {
+        expectedBloomFilter.insertHash(expectedBloomFilter.hash(Binary.fromString(elements[columnIndex])));
+      } else {
+        expectedBloomFilter.insertHash(expectedBloomFilter.hash(Integer.parseInt(elements[columnIndex])));
+      }
+    }
+    return expectedBloomFilter;
+  }
+
+
+  private void waitForJob(Job job) throws Exception {
+    job.submit();
+    while (!job.isComplete()) {
+      sleep(100);
```

Comment:
Maybe some timeout here? What happens if the job doesn't end at all?

Reply:
https://github.com/junit-team/junit4/wiki/Timeout-for-tests#timeout-parameter-on-test-annotation-applies-to-test-method

Reply:
Good question. What I'm trying to do is quite different than the timeout of Juint. In fact, I'm waiting for the job to finish before moving to the next part where I check the bloom filter.  

Since we know the content of the mapper we can say that the job will end. Therefore, the job either will succeed or fail.

Btw, thanks for sharing the link.